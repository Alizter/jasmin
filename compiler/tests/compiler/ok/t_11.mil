// macro expansion

param n : u64;
param m : u64;

extern fn test() {

  cf            : reg bool;
  ptr           : reg u64;
  v_14           : stack u64[n];
  v_0           : stack u64[n];
  v_inner       : stack u64[n];
  v_inner_outer : stack u64[n];
  v_inner_inner : stack u64[m];
  v_dead        : stack u64[m];
  i, j          : inline u64;

  if cf {
    v_0[0] = 0; 
    for i in 10..$n {
      if $n = 15 {
        v_14[$n - 1] += 1;
        v_inner[i] = i;
        for j in 5..$m {
          v_inner_outer[i] = MEM[ptr + i*8];
          v_inner_inner[j] = j;
          MEM[ptr + j*8] = j;
        }
      } else {
        v_dead[$n - 1] -= 1;
      }
    }
  }
}

/*
START:CMD
ARG="renumber_fun_unique,typecheck,expand[test][n=15,m=7],renumber_reuse,print[renum][]"
END:CMD
*/
