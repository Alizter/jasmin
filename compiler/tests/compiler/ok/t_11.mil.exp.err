Processed file tests/compiler/ok/t_11.mil
