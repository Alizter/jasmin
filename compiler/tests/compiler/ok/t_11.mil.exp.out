renumbering identifiers in all function(s)
   REMOVED_TIME
type checking module in all function(s)
   REMOVED_TIME
expanding macros in ``test'' function(s)
   REMOVED_TIME
renumbering identifiers in all function(s)
   REMOVED_TIME
>> renum:
extern fn test() {
  cf : reg bool
  ptr : reg u64
  v_0 : stack u64[15]
  v_14 : stack u64[15]
  v_inner : stack u64[15]
  v_inner_inner : stack u64[7]
  v_inner_outer : stack u64[15]
  if cf {
    v_0[0] = 0;
    v_14[14] += 1;
    v_inner[10] = 10;
    v_inner_outer[10] = MEM[ptr + 80];
    v_inner_inner[5] = 5;
    MEM[ptr + 40] = 5;
    v_inner_outer[10] = MEM[ptr + 80];
    v_inner_inner[6] = 6;
    MEM[ptr + 48] = 6;
    v_14[14] += 1;
    v_inner[11] = 11;
    v_inner_outer[11] = MEM[ptr + 88];
    v_inner_inner[5] = 5;
    MEM[ptr + 40] = 5;
    v_inner_outer[11] = MEM[ptr + 88];
    v_inner_inner[6] = 6;
    MEM[ptr + 48] = 6;
    v_14[14] += 1;
    v_inner[12] = 12;
    v_inner_outer[12] = MEM[ptr + 96];
    v_inner_inner[5] = 5;
    MEM[ptr + 40] = 5;
    v_inner_outer[12] = MEM[ptr + 96];
    v_inner_inner[6] = 6;
    MEM[ptr + 48] = 6;
    v_14[14] += 1;
    v_inner[13] = 13;
    v_inner_outer[13] = MEM[ptr + 104];
    v_inner_inner[5] = 5;
    MEM[ptr + 40] = 5;
    v_inner_outer[13] = MEM[ptr + 104];
    v_inner_inner[6] = 6;
    MEM[ptr + 48] = 6;
    v_14[14] += 1;
    v_inner[14] = 14;
    v_inner_outer[14] = MEM[ptr + 112];
    v_inner_inner[5] = 5;
    MEM[ptr + 40] = 5;
    v_inner_outer[14] = MEM[ptr + 112];
    v_inner_inner[6] = 6;
    MEM[ptr + 48] = 6;
  }
}


total transformation time: REMOVED_TIME

total time (with save/print): REMOVED_TIME
